components/tools/OmeroPy/src/omero/plugins/web.py

```diff
@@ -210,20 +221,40 @@ def _set_apache_wsgi(self, d, settings):
         d["OMEROPYTHONROOT"] = self._get_python_dir()
         d["OMEROFALLBACKROOT"] = self._get_fallback_dir()
 
+    def _fastcgi_deprecation(self, settings):
+        if settings.APPLICATION_SERVER in settings.FASTCGI_TYPES:
+            self.ctx.err(
+                "WARNING: FastCGI support is deprecated and will be removed"
+                " in OMERO 5.2. Install gunicorn and update config.")
+
+    def _assert_config_argtype(self, argtype, settings):
+        mismatch = False
+        if (settings.APPLICATION_SERVER in settings.WSGI_TYPES and
+                "wsgi" not in argtype):
+            mismatch = True
+        if settings.APPLICATION_SERVER in ("development",):
+            mismatch = True
+        if (settings.APPLICATION_SERVER in settings.FASTCGI_TYPES and
+                argtype not in ("nginx", "nginx-development",
+                                "apache", "apache-fcgi")):
+            mismatch = True
+        if mismatch:
+            self.ctx.die(1, ("ERROR: configuration mismatch. "
```

Comment:
I think the reason for that error is to prevent people from misconfiguration web app. It should be strict what is allowed and when. We already had enough misconfiguration issues on the forum because default params has changed in 5.1 and copy config file as stated in the doc

```
cp OMERO.server-old/etc/grid/config.xml OMERO.server/etc/grid
```

There should not be an option to omit. Lets keep it simple.